app/src/main/java/com/example/sharewhatyoucanproject/login/LoginActivity.kt

```diff
@@ -0,0 +1,110 @@
+package com.example.sharewhatyoucanproject.login
+
+import android.app.ProgressDialog
+import android.content.Intent
+import android.os.Bundle
+import android.provider.Settings
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.ViewModelProvider
+import com.example.sharewhatyoucanproject.DrawerActivity
+import com.example.sharewhatyoucanproject.SelectUserActivity
+import com.example.sharewhatyoucanproject.databinding.ActivityLoginBinding
+import com.example.sharewhatyoucanproject.models.UserType
+import com.example.sharewhatyoucanproject.utils.editSharedPreferencesSelector
+import com.example.sharewhatyoucanproject.utils.showToast
+
+class LoginActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityLoginBinding
+    lateinit var loginViewModel: LoginViewModel
+    lateinit var progressDialog: ProgressDialog
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityLoginBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        progressDialog = ProgressDialog(this)
+        progressDialog.setTitle("Please Wait")
+        supportActionBar?.hide()
+
+        loginViewModel = ViewModelProvider(
+            this,
+            LoginViewModelFactory(),
+        )[LoginViewModel::class.java]
+
+        val type = intent.getIntExtra("type", 0)
+
+        val typevar1 = UserType.values()[type]
+
+        loginViewModel.type = typevar1
+
+        loginViewModel.checkResult.observe(this) { result ->
+            result?.let {
+                progressDialog.dismiss()
+                // this.showToast(result)
+                // TODO show toast
+                loginViewModel.checkResult.value = null
+            }
+        }
+
+        loginViewModel.loginResult.observe(this) { loginResult ->
+            loginResult?.let {
+                progressDialog.dismiss()
+                if (loginResult == "success") {
```

Comment:
I think you could probably combine all of these result types into one sealed class, for example:

```
sealed class AuthenticationResult {
    data class Fail(val message: String, val error: Throwable): AuthenticationResult()
    object LoginSuccess : AuthenticationResult()
    object SignUpSuccess : AuthenticationResult()
}
```

This way you can rely on a strong type instead of strings

usage would be something like:
```
      loginViewModel.authResult.observe(this) { result ->
            when (result) {
                is AuthenticationResult.Fail -> // do something with error
                AuthenticationResult.LoginSuccess -> // send to DrawerActivity
                AuthenticationResult.SignUpSuccess -> // send to select user activity or drawer activity based on type
            }
        }
```